prost-build/src/code_generator.rs

```diff
@@ -15,25 +15,21 @@ use prost_types::{
 };
 
 use crate::ast::{Comments, Method, Service};
-use crate::extern_paths::ExternPaths;
+use crate::context::Context;
 use crate::ident::{strip_enum_prefix, to_snake, to_upper_camel};
-use crate::message_graph::MessageGraph;
-use crate::Config;
 
 mod c_escaping;
 use c_escaping::unescape_c_escape_string;
 
 mod syntax;
 use syntax::Syntax;
 
-pub struct CodeGenerator<'a> {
-    config: &'a mut Config,
+pub struct CodeGenerator<'a, 'b> {
+    context: &'a mut Context<'b>,
```

Comment:
I don't understand why some fields are in `CodeGenerator`, `Context` or `Config`.  For example, `depth` also feels like it could be part of `Context`.

Can you explain when a field should be in each of those structs?